pkg/apiserver/rest/dualwriter_mode1.go

```diff
@@ -27,36 +33,191 @@ func NewDualWriterMode1(legacy LegacyStorage, storage Storage) *DualWriterMode1
+	var method = "delete"
+
+	startLegacy := time.Now().UTC()
+	res, async, err := d.Legacy.Delete(ctx, name, deleteValidation, options)
+	if err != nil {
+		log.Error(err, "unable to delete object in legacy storage")
+		d.recordLegacyDuration(true, mode1, name, method, startLegacy)
+		return res, async, err
+	}
+	d.recordLegacyDuration(false, mode1, name, method, startLegacy)
+
+	go func() {
+		startStorage := time.Now().UTC()
+		ctx, _ := context.WithTimeoutCause(ctx, time.Second*10, errors.New("storage delete timeout"))
+		_, _, err := d.Storage.Delete(ctx, name, deleteValidation, options)
+		d.recordStorageDuration(err != nil, mode1, name, method, startStorage)
+	}()
+
+	return res, async, err
 }
 
 // DeleteCollection overrides the behavior of the generic DualWriter and deletes only from LegacyStorage.
 func (d *DualWriterMode1) DeleteCollection(ctx context.Context, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions, listOptions *metainternalversion.ListOptions) (runtime.Object, error) {
-	ctx = klog.NewContext(ctx, d.Log)
-	return d.Legacy.DeleteCollection(ctx, deleteValidation, options, listOptions)
+	log := d.Log.WithValues("kind", options.Kind, "resourceVersion", listOptions.ResourceVersion)
+	ctx = klog.NewContext(ctx, log)
+	var method = "delete-collection"
+
+	startLegacy := time.Now().UTC()
+	res, errLegacy := d.Legacy.DeleteCollection(ctx, deleteValidation, options, listOptions)
+	if errLegacy != nil {
+		log.Error(errLegacy, "unable to delete collection in legacy storage")
+		d.recordLegacyDuration(true, mode1, options.Kind, method, startLegacy)
+	}
+	d.recordLegacyDuration(errLegacy != nil, mode1, options.Kind, method, startLegacy)
+
+	go func() {
+		startStorage := time.Now().UTC()
+		ctx, _ := context.WithTimeoutCause(ctx, time.Second*10, errors.New("storage deletecollection timeout"))
+		_, err := d.Storage.DeleteCollection(ctx, deleteValidation, options, listOptions)
+		d.recordStorageDuration(err != nil, mode1, options.Kind, method, startStorage)
+	}()
+
+	return res, errLegacy
 }
 
 func (d *DualWriterMode1) Update(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
-	ctx = klog.NewContext(ctx, d.Log)
-	return d.Legacy.Update(ctx, name, objInfo, createValidation, updateValidation, forceAllowCreate, options)
+	log := d.Log.WithValues("name", name, "kind", options.Kind)
+	ctx = klog.NewContext(ctx, log)
+	var method = "update"
+
+	startLegacy := time.Now().UTC()
+	res, async, errLegacy := d.Legacy.Update(ctx, name, objInfo, createValidation, updateValidation, forceAllowCreate, options)
+	if errLegacy != nil {
+		log.Error(errLegacy, "unable to update in legacy storage")
+		d.recordLegacyDuration(true, mode1, name, method, startLegacy)
+	}
+	d.recordLegacyDuration(false, mode1, name, method, startLegacy)
```

Comment:
`name` here is going to be the complete k8s name, we just want to pass the Kind (and only call recordLegacyDuration once)
```suggestion
		d.recordLegacyDuration(true, mode1, options.Kind, method, startLegacy)
	} else {
		d.recordLegacyDuration(false, mode1, options.Kind, method, startLegacy)
	}
```

Reply:
oh, I wasn't aware this would be the full k8s object! Should I also change in the other modes?

Reply:
Yep, we'll need to update that everywhere